x-pack/plugin/ent-search/qa/full-cluster-restart/javaRestTest/java/org/elasticsearch/entsearch/FullClusterRestartIT.java

```diff
@@ -0,0 +1,115 @@
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package org.elasticsearch.entsearch;
+
+import com.carrotsearch.randomizedtesting.annotations.Name;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.local.distribution.DistributionType;
+import org.elasticsearch.test.rest.ObjectPath;
+import org.elasticsearch.upgrades.FullClusterRestartUpgradeStatus;
+import org.elasticsearch.upgrades.ParameterizedFullClusterRestartTestCase;
+import org.junit.ClassRule;
+
+import java.io.IOException;
+import java.util.List;
+
+public class FullClusterRestartIT extends ParameterizedFullClusterRestartTestCase {
+
+    // Legacy name we used for ILM policy configuration in versions prior to 8.11.
+    private static final String EVENT_DATA_STREAM_LEGACY_ILM_POLICY_NAME = "behavioral_analytics-events-default_policy";
+
+    @ClassRule
+    public static final ElasticsearchCluster cluster = ElasticsearchCluster.local()
+        .distribution(DistributionType.DEFAULT)
+        .version(getOldClusterTestVersion())
+        .nodes(2)
+        .setting("xpack.security.enabled", "false")
+        .module("x-pack-ent-search")
+        .build();
+
+    public FullClusterRestartIT(@Name("cluster") FullClusterRestartUpgradeStatus upgradeStatus) {
+        super(upgradeStatus);
+    }
+
+    @Override
+    protected ElasticsearchCluster getUpgradeCluster() {
+        return cluster;
+    }
+
+    public void testBehavioralAnalyticsDataRetention() throws Exception {
+
+        String legacyAnalyticsCollectionName = "oldstuff";
+        String newAnalyticsCollectionName = "newstuff";
+
+        if (isRunningAgainstOldCluster()) {
+            // Create an analytics collection
+            Request legacyPutRequest = new Request("PUT", "_application/analytics/" + legacyAnalyticsCollectionName);
+            assertOK(client().performRequest(legacyPutRequest));
+
+            // Validate that ILM lifecycle is in place
+            assertBusy(() -> testLegacyDataRetentionPolicy(legacyAnalyticsCollectionName));
+        } else {
+            // Create a new analytics collection
+            Request putRequest = new Request("PUT", "_application/analytics/" + newAnalyticsCollectionName);
+            assertOK(client().performRequest(putRequest));
+
+            // Validate that NO ILM lifecycle is in place and we are using DLM instead.
+            assertBusy(() -> testDlmDataRetentionPolicy(newAnalyticsCollectionName));
+
+            // Validate that the existing analytics collection created with an older version is still using ILM
+            assertBusy(() -> testLegacyDataRetentionPolicy(legacyAnalyticsCollectionName));
+        }
+    }
+
+    private void testLegacyDataRetentionPolicy(String analyticsCollectionName) throws IOException {
```

Comment:
I usually try to not use `test` prefix for method that are not tests.
I would recommend to rename the method into `assertLegacyDataRetentionPolicy` (which make sense since used into a `assertBusy`

Reply:
Good catch, will change 👍